### ISSUE TITLE
RecurrenceRule returns wrong BYMONTH part value

### ISSUE THREAD

**Issue by yermolovich1987** [dmfs/lib-recur#98]
Hi all. Recently I have found a strange behaviour that seems like a bug. When we build a RecurrentRule with BYMONTH part and then try to get the value of BYMONTH part from the result object via `rule.getByPart(RecurrenceRule.Part.BYMONTH)` - it returns the values that are reduced on 1. At the same time, if we will try to convert the object back to the RRULE string or calculate the next occurrence date - all is working correctly. As I can see, this happens because when GregorianCalendarMetrics parses BYMONTH for internal purposes it decrease the month and then increase back during serialization back to String. At the same time, from API point of view this seems confusing since I expect to receive always the correct values - both when get a particular part via direct method call or when convert to result String.
The sample code to demonstrate the issue:
```
var ruleToTest = "FREQ=WEEKLY;BYMONTH=11;COUNT=1"; 
var rule = new RecurrenceRule(ruleToTest);
var iterator = rule.iterator(DateTime.parse("20210701T120000Z"));
System.out.println(iterator.nextDateTime());
System.out.println(rule.getByPart(RecurrenceRule.Part.BYMONTH));
System.out.println(rule.toString());
```
The output will look like:
```
20211104T120000Z
[10]
FREQ=WEEKLY;BYMONTH=11;COUNT=1
```
As you can see, the calculated next date and the RRULE string contains the correct month - 11. But the rule.getByPart(RecurrenceRule.Part.BYMONTH) returns the value 10. 

Is this an intended behaviour and is there any proper way to receive the correct value of the BYMONTH part?
Thanks!

**Comment by dmfs** [comment]
Right, sounds like a bug indeed. Shouldn't be hard to fix though. I'll check that.

**Comment by dmfs** [comment]
After looking into the code again, it realized this was actually intended behaviour. RFC 5545 (and therefore RRULE) use 1-based month numbers, whereas Java's Calendar class (which served as the model for the implementation back then, when I started this) uses 0-based month numbers.
IIRC, the idea was to separate String representation and API, just like the Calendar class does. So to retrieve the actual month number you have to add `1`. For calendars which have leap months it would be more difficult though, but that's not supported yet. I guess the API needs to be reworked a bit.

**Comment by yermolovich1987** [comment]
Hi Marten. Thanks for response and for library! I think that you are right and in that case better to update an API so that rule.getByPart(RecurrenceRule.Part.BYMONTH) returns the same month number as configured in the original rule/string representation.

**Comment by GoldenSoju** [comment]
Hi all~!
I was wondering if there are any updates on this issue?
